Tests/Microsoft.AppCenter.Test.Functional/AppCenter/AppCenterCoreTest.cs

```diff
@@ -0,0 +1,203 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace Microsoft.AppCenter.Test.Functional.AppCenter
+{
+    using Analytics = Microsoft.AppCenter.Analytics.Analytics;
+    using AppCenter = Microsoft.AppCenter.AppCenter;
+
+    public class AppCenterCoreTest : IDisposable
+    {
+
+        // Before
+        public AppCenterCoreTest()
+        {
+            Utils.deleteDatabase();
+        }
+
+        // After
+        public void Dispose()
+        {
+            // Let pending SDK calls be completed.
+            Task.Delay(3000).Wait();
+        }
+
+        [Fact]
+        public async Task EnableDisableTest()
+        {
+            // Start App Center.
+            AppCenter.UnsetInstance();
+            Analytics.UnsetInstance();
+            AppCenter.LogLevel = LogLevel.Verbose;
+            AppCenter.Start(Config.resolveAppsecret(), typeof(Analytics));
+
+            // Disable Appcenter.
+            await AppCenter.SetEnabledAsync(false);
+
+            // Verify disabled.
+            var isEnabled = await AppCenter.IsEnabledAsync();
+            var isEnabledAnalytics = await Analytics.IsEnabledAsync();
+            Assert.False(isEnabled);
+            Assert.False(isEnabledAnalytics);
+
+            // Restart SDK.
+            AppCenter.UnsetInstance();
+            Analytics.UnsetInstance();
+            AppCenter.LogLevel = LogLevel.Verbose;
+            AppCenter.Start(Config.resolveAppsecret(), typeof(Analytics));
+
+            // Verify disabled.
+            var isEnabled2 = await AppCenter.IsEnabledAsync();
+            var isEnabledAnalytics2 = await Analytics.IsEnabledAsync();
+            Assert.False(isEnabled2);
+            Assert.False(isEnabledAnalytics2);
+
+            // Enable SDK.
+            await AppCenter.SetEnabledAsync(true);
+
+            // Verify enabled.
+            var isEnabled3 = await AppCenter.IsEnabledAsync();
+            var isEnabledAnalytics3 = await Analytics.IsEnabledAsync();
+            Assert.True(isEnabled3);
+            Assert.True(isEnabledAnalytics3);
+
+            // Restart SDK.
+            AppCenter.UnsetInstance();
+            Analytics.UnsetInstance();
+            AppCenter.LogLevel = LogLevel.Verbose;
+            AppCenter.Start(Config.resolveAppsecret(), typeof(Analytics));
+
+            // Verify enabled.
+            var isEnabled4 = await AppCenter.IsEnabledAsync();
+            var isEnabledAnalytics4 = await Analytics.IsEnabledAsync();
+            Assert.True(isEnabled4);
+            Assert.True(isEnabledAnalytics4);
+
+            // Let pending SDK calls be completed, we have a lot of "startService" calls.
+            Task.Delay(5000).Wait();
```

Comment:
Verify start service is sent instead so that you don't need this. Use the same waiting on payload code used for event in analytics tests.

Reply:
Will be fixed separately